rewrite-pattern.js

```diff
@@ -387,32 +414,36 @@ const computeCharacterClass = (characterClassItem, regenerateOptions) => {
 	}
 
 	if (characterClassItem.negative && data.maybeIncludesStrings) {
-		throw new SyntaxError("Cannot negate set containing strings");
+		throw new SyntaxError('Cannot negate set containing strings');
 	}
 
 	return data;
 }
 
-const processCharacterClass = (characterClassItem, regenerateOptions) => {
+const processCharacterClass = (
+	characterClassItem,
+	regenerateOptions,
+	computed = computeCharacterClass(characterClassItem, regenerateOptions),
+) => {
 	const negative = characterClassItem.negative;
-	const { singleChars, transformed, longStrings } = computeCharacterClass(characterClassItem, regenerateOptions);
+	const { singleChars, transformed, longStrings } = computed;
 	if (transformed) {
 		const setStr = singleChars.toString(regenerateOptions);
 
 		if (negative) {
 			if (config.useUnicodeFlag) {
-				update(characterClassItem, `[^${setStr[0] === "[" ? setStr.slice(1, -1) : setStr}]`)
+				update(characterClassItem, `[^${setStr[0] === '[' ? setStr.slice(1, -1) : setStr}]`)
 			} else {
 				update(characterClassItem, `(?!${setStr})[\\s\\S]`)
 			}
 		} else {
-			const hasEmptyString = longStrings.has("");
+			const hasEmptyString = longStrings.has('');
 			const pieces = Array.from(longStrings).sort((a, b) => b.length - a.length);
-			if (setStr !== "[]" || longStrings.size === 0) {
+			if (setStr !== '[]' || longStrings.size === 0) {
 				pieces.splice(pieces.length - (hasEmptyString ? 1 : 0), 0, setStr);
 			}
 
-			update(characterClassItem, pieces.join("|"));
+			update(characterClassItem, pieces.join('|'));
```

Comment:
If any of the `pieces` include the raw character `|`, we need to escape those as `\|`.

Reply:
Oh actually, this already works. The `|` in `pieces` must already be escaped (for example, in `[\q{a\|b|cd}]`) otherwise it's already interpreted as a separator.